src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```
@@ -63,16 +62,46 @@ Copyright (c) .NET Foundation. All rights reserved.
     </BuildOutputInPackage>
   </ItemDefinitionGroup>
 
-  <Target Name="_GetOutputItemsFromPack"
-          Returns="@(_OutputPackItems)">
+  <PropertyGroup>
+    <PackageOutputPath Condition="'$(PackageOutputPath)' == ''">$(BaseOutputPath)$(Configuration)\</PackageOutputPath>
```

```suggestion
    <PackageOutputPath Condition="'$(PackageOutputPath)' == ''">$(OutputPath)</PackageOutputPath>
```

---

This **change** is the **actual PR**.

---

The change is scope of the PackageOutputPath property, so that installing the pack sdk into non-sdk PackageReference based projects enables to `msbuild /t:pack` and not get the failure about a missing PackageOutputPath.

---

I just tested on non-style projects, it doesn't work since `BaseOutputPath` is not there in common targets. Either we can change to `OutputPath` when base is not available or we could add base to common targets.

---

I don't have an answer, but I wanted to thank you for testing on non-SDK style projects. This isn't the easiest change to test, so I really appreciate you made the effort of testing different scenarios.

---

```suggestion
    <PackageOutputPath Condition="'$(PackageOutputPath)' == '' AND '$(BaseOutputPath)' == ''">$(OutputPath)</PackageOutputPath>
    <PackageOutputPath Condition="'$(PackageOutputPath)' == '' AND '$(BaseOutputPath)' != ''">$(BaseOutputPath)$(Configuration)\</PackageOutputPath>
```

---

Well, I do have a **lot** of framework projects!! 😎

---

How about...

This?
```suggestion
    <PackageOutputPath Condition="'$(PackageOutputPath)' == '' AND '$(BaseOutputPath)' == ''">$(OutputPath)</PackageOutputPath>
    <PackageOutputPath Condition="'$(PackageOutputPath)' == '' AND '$(BaseOutputPath)' != ''">$(BaseOutputPath)$(Configuration)\</PackageOutputPath>
```

OR This?
```suggestion
    <PackageOutputPath Condition="'$(PackageOutputPath)' == '' AND '$(UsingMicrosoftNETSdk)' != 'true'">$(OutputPath)</PackageOutputPath>
    <PackageOutputPath Condition="'$(PackageOutputPath)' == '' AND '$(UsingMicrosoftNETSdk)' == 'true'">$(BaseOutputPath)$(Configuration)\</PackageOutputPath>
```

I prefer the first one, since it depends on the property itself. The second one will work (even from v1) but what if they use a custom Sdk (I do) with similar defaults as .NET Sdk, then the latter fails.

---

Apologies for the delay. 

I am not sure we are all on the same page on the problem we are trying to fix. 

I have replied to the original issue because I think the discussion is best had there. https://github.com/NuGet/Home/issues/9234#issuecomment-604114471

Specifically, I cannot accept this change because in non-SDK based projects it will end up a directory structure as follows: 

App\
  bin\Debug\*.dll, *.exe
  Debug\*.nupkg
  obj\Debug\*.nuspec, *.dll

In SDK based projects the structure is and should remain:

bin\Debug\*.nupkg
bin\Debug\targetFrameworkName1\*.dll
bin\Debug\targetFrameworkName2\*.dll
obj\Debug\*.nuspec
obj\Debug\targetFrameworkName1\*.dll
obj\Debug\targetFrameworkName2\*.dll

The outputs should mimic one another.

---

https://github.com/NuGet/Home/issues/9234#issuecomment-604138764